zerver/tests/test_events.py

```diff
@@ -4038,13 +4038,11 @@ def test_change_realm_property(self) -> None:
         for prop in Realm.REALM_PERMISSION_GROUP_SETTINGS:
             with self.settings(SEND_DIGEST_EMAILS=True):
                 self.do_set_realm_permission_group_setting_test(prop)
-
-        for prop in Realm.REALM_PERMISSION_GROUP_SETTINGS_WITH_NEW_API_FORMAT:
             if Realm.REALM_PERMISSION_GROUP_SETTINGS[prop].require_system_group:
                 # Anonymous system groups aren't relevant when
                 # restricted to system groups.
                 continue
-            with self.settings(SEND_DIGEST_EMAILs=True):
+            with self.settings(SEND_DIGEST_EMAILS=True):
                 self.do_set_realm_permission_group_setting_to_anonymous_groups_test(prop)
```

Comment:
Do we need to run both `do_set_realm_permission_group_setting_test` and `do_set_realm_permission_group_setting_to_anonymous_groups_test`, or is the former test unnecessary if we are doing the anonymous groups case? 

I'm wondering if this should be an if/else clause rather than doing the `continue`.

Reply:
The test for anonymous groups will test setting the value to a system group only for the default value while `do_set_realm_permission_group_setting_test` will try to set it for all allowed system groups. The test for anonymous groups has one test for a system group, so unless there is a reason to try all system groups for those settings, we can do it in an if/else way. 

cc. @sahil839 if there's a reason in your mind to check against all system groups.

Reply:
I think the test for all system groups was mainly to make sure that different fields in `GroupPermissionSetting` like `allow_nobody_group`, `allow_everyone_group`, etc. work as expected.

Reply:
Added 8628c9625b928c5de53d44b20e39d5789829051a to rename the test and just keep it limited to testing those settings.